### PR TITLE
Round 11 - Satire Satyrs

### PR DESCRIPTION
**Pull request by Shushishtok**
* Adds round 11: Satire Satyrs, which includes evil Necrocursers that apply Unholy Curse on you, slowing you down for a short period.
* Slightly reworked how ShouldAbilityBeCast() works. It now also may pass a target and a location argument for checking purposes. Those are based on the type of the ability cast, in the following manner:
    - Unit target abilities will supply the target the ability is cast on, but not location.
    - No target abilities will not supply a target, and will supply the current location of the parent as the location.
    - Point abilities will not supply a target, and will supply the chosen position to cast the ability at.
    - Other ability types added later might have additional logics provided later.
* Fixed a bug where OnWaveComplete() is called twice when ending waves via dev commands.
* Fixed a bug where your health and mana are set to 50% of your max values instead of being healed by those values.
* Attempts to fix annoying prints in console, telling you that the unit being ordered to execute an order is dead, by adding IsAlive checks everywhere (there already was one at the top of each interval thinks, but that seems to check too early or something). Maybe there's a better way to do this though.

### CODE REVIEW

File: game/scripts/vscripts/core/interfaces.d.ts [hunk 1]
```diff
@@ -9,7 +9,23 @@ interface CDOTA_BaseNPC_Hero {
 }
 
 interface CDOTABaseAbility {
+	/**
+	 * Implemented in wave-based Unit target abilities. Determines the target of an ability to be cast based on conditions set.
+	 * @returns Either a target to be selected on abilities, or undefined if no valid target was found and the ability shouldn't be cast.
+	 */
 	GetAbilityTargetLogic(): CDOTA_BaseNPC | undefined;
```
Comment (Nibuja05) [inline game/scripts/vscripts/core/interfaces.d.ts:16]: Name is really misleading. From the name I'd expect it to return `WaveUnitAILogicType`, but it does return a unit instead.
Maybe rename it to `GetAbilityLogicTarget`?

File: game/scripts/vscripts/core/interfaces.d.ts [hunk 2]
```diff
@@ -9,7 +9,23 @@ interface CDOTA_BaseNPC_Hero {
...
+
+	/**
+	 * Implemented in wave-based Point abilities. Determines the position that the ability should be cast based on conditions set.
+	 * @returns Either a position (vector) to cast the ability on, or undefined if no valid position was found and the ability shouldn't be cast.
+	 */
 	GetAbilityPositionLogic(): Vector | undefined;
```
Comment (Nibuja05) [inline game/scripts/vscripts/core/interfaces.d.ts:22]: Same here, maybe rename to `GetAbilityLogicPosition`?

File: game/scripts/vscripts/core/wave_manager.ts [hunk 3]
```diff
@@ -471,10 +471,25 @@ export class WaveManager {
...
 	 */
 	private OnWaveComplete() {
+		if (!this.started) return;
+		this.started = false;
+
 		// Send event
 		let player = PlayerResource.GetPlayer(this.playerID)!;
 		CustomGameEventManager.Send_ServerToPlayer(player, "wave_complete", {});
 
+		// Find all enemy units and kill them. This will usually only proc
```
Comment (Nibuja05) [inline game/scripts/vscripts/core/wave_manager.ts:481]: *This will usually only proc*... Somethings missing here 😅

**Review by Nibuja05 — COMMENTED** [review]
Looking super good!
Just some very minor things

File: game/scripts/vscripts/core/interfaces.d.ts (hunk 1 above)
Comment (Shushishtok) [reply game/scripts/vscripts/core/interfaces.d.ts:16]: Done!

File: game/scripts/vscripts/core/interfaces.d.ts (hunk 2 above)
Comment (Shushishtok) [reply game/scripts/vscripts/core/interfaces.d.ts:22]: Done!

File: game/scripts/vscripts/core/wave_manager.ts (hunk 3 above)
Comment (Shushishtok) [reply game/scripts/vscripts/core/wave_manager.ts:481]: Fixed!

**Review by Nibuja05 — APPROVED** [review]

**Pull request merged**